src/System.Device.Gpio/System/Device/Gpio/GpioController.cs

```diff
@@ -75,14 +85,23 @@ private int GetLogicalPinNumber(int pinNumber)
         /// <param name="pinNumber">The pin number in the controller's numbering scheme.</param>
         public void OpenPin(int pinNumber)
         {
-            int logicalPinNumber = GetLogicalPinNumber(pinNumber);
-            if (_openPins.Contains(logicalPinNumber))
+            if (IsPinOpen(pinNumber))
             {
-                throw new InvalidOperationException($"Pin {logicalPinNumber} is already open.");
+                throw new InvalidOperationException($"Pin {pinNumber} is already open.");
             }
 
+            OpenPinCore(pinNumber);
+            _openPins.Add(pinNumber);
```

Comment:
Should the opertaion of adding it to the openPins list live instead on OpenPinCore method? Or is the idea of having it here so that this part will never be overriden by subclasses?

Reply:
Yes, that was exactly the reason to do it here. Getting this wron in an overriden class would confuse the drivers significantly. And the list of open pins is not public.